Admin: Remove the site card and move the badges to the site dropdown in the master bar

youknowriad opened the pull request:
Related to #98900 
Alternative to #99536

##  Summary

This PR implements a significant UI improvement to the WordPress.com admin interface by removing the redundant site card from the sidebar and moving site status badges to the site dropdown in the master bar. This change creates a cleaner, more focused interface while maintaining all necessary functionality and information. It does make the `/view` pages inaccessible. It remains to be seen how useful these pages are.

## Changes

  - Created a new SiteNotices component that maintains all the important notices previously shown with the site card
  - Modified the sidebar to use the new SiteNotices component instead of CurrentSite
  - Enhanced the masterbar to display site status information in a dropdown menu
  - Added styling for the site badges in the masterbar (both multi site and single site)
  - Added conditional logic to display plan information appropriately

  Testing Instructions

  1. View the WordPress.com admin interface with this PR applied
  2. Verify that site information (previously in the sidebar card) now appears in the site dropdown menu in the top bar
  3. Test with various site types (Private, Coming Soon, sites with updates, etc.) to ensure all badges display correctly
  4. Verify that all important site notices still appear at the top of the sidebar

  Screenshots

<img width="317" alt="Screenshot 2025-03-04 at 12 08 58 PM" src="https://github.com/user-attachments/assets/0b5afe8a-f677-4756-8844-8a1956bc8237" />
<img width="433" alt="Screenshot 2025-03-04 at 12 09 17 PM" src="https://github.com/user-attachments/assets/e02f3ec7-ea4c-45dd-8ae8-b92199445bae" />

  Motivation and Context

  This change improves the user experience by:
  - Reducing UI clutter in the sidebar
  - Consolidating site information in a logical location
  - Creating more vertical space for navigation options
  - Maintaining all important status information and notifications

youknowriad committed:
Admin: Remove the site card and move the badges to the site dropdown in the master bar

youknowriad commented on client/layout/masterbar/item.tsx:
```diff
@@ -137,6 +137,7 @@ class MasterbarItem extends Component< MasterbarItemProps > {
 								{ item.label }
 							</a>
 						) }
+						{ ! item.onClick && ! item.url && item.label }
```
This basically adds support for non clickable items to the master bar menus.

ciampo replied:
It may be better to wrap them in a `<span />` for better "consistency" and potentially less layout differences with the other item types (`<button />` and `<a />`) ?

youknowriad commented on client/layout/masterbar/logged-in.jsx:
```diff
@@ -400,6 +407,139 @@ class MasterbarLoggedIn extends Component {
+		}
+
+		return badges.length > 0 ? badges : null;
+	}
```
It might be worth extracting these gadgets to a a `SiteBadges` component and reuse in both places.

youknowriad commented on client/layout/masterbar/style.scss:
```diff
@@ -936,6 +936,105 @@ body.is-mobile-app-view {
+	}
+}
+
+.masterbar__site-badge {
```
@ciampo I know there were recent discussions (and maybe implementation) of badge components in Calypso, This is adhoc but I'm happy to use the right component (I don't know if there's any that matches what we want to do here).

ciampo replied:
Hey Riad, this is the current situation AFAIK:

- It looks like there is [a `Badge` component from `@automattic/components`](https://github.com/Automattic/wp-calypso/tree/d2f640d00a8f4b4782dca1b9153454ed3fc30c43/packages/components/src/badge), but I'm not sure how "up to date" with design requirements it is.
- We had also started working on [a `Badge` component in `@wordpress/components`](https://wordpress.github.io/gutenberg/?path=/docs/components-badge--docs), but that component is still WIP and exported as an experimental API.
- Because of that, folks [have created the counterpart in Calypso](https://github.com/Automattic/wp-calypso/issues/99343) by `unlock`-ing and exposing the private API.

Design-wise, the badge that I see from the screenshots doesn't align with any of the above badges, though — so we may want to double-check the design specs and see if we can re-use the current designs

(cc @mirka )

youknowriad commented on client/layout/sidebar/style.scss:
```diff
@@ -3,6 +3,7 @@
 	// margins and paddings.
 	margin: 0;
 	padding: 0;
+	padding-top: 6px;
```
I found that this small padding on top of the sidebar makes things a bit better when there's no upsell or notice on top of the menu.

nuriapenya reviewed (changes requested):
Is there a reason to change the background colour on the global level? I guess is to match the UI of that area, but since we are reusing the wp-admin bar, I'd suggest to use the dark background always. Otherwise is quite inconsistent:
![screen 2025-03-04 at 13 03 30](https://github.com/user-attachments/assets/46d7b069-9900-44cc-a123-d9b771e15857)

I'd also suggest removing the little dot (yellow, blue) since we haven't defined a color guidelines and right now is arbitrary? (cc. @jameskoster)
 
![screen 2025-03-04 at 13 07 19](https://github.com/user-attachments/assets/7257187d-5996-4eed-8b04-77fefefecef2)

Do we need the "Updates" badge there too? I think this would belong to the updates tab, right?

About the plan badge: I like it there, but as a user I'd expect to be able to click on it and go the plans page, it's a good subtle way to upsell. wdyt @richtabor ? I'd remove the dot with the colour though.

youknowriad commented:
> Is there a reason to change the background colour on the global level? I guess is to match the UI of that area, but since we are reusing the wp-admin bar, I'd suggest to use the dark background always. Otherwise is quite inconsistent:

I agree that we should ideally be consistent but If we do change the background in the global section, we should change it consistently. Changing it only for the new dropdown, wouldn't feel great IMO.

fushar commented:
> Is there a reason to change the background colour on the global level? I guess is to match the UI of that area, but since we are reusing the wp-admin bar, I'd suggest to use the dark background always. Otherwise is quite inconsistent:

@nuriapenya: It's unfortunately already inconsistent now. The global view always uses the "white" color scheme, while in the site-specific screens (e.g. My Home), we are using the site's color scheme. So it's not something introduced in this PR.

nuriapenya commented:
Thank you both! 🙏🏼 I understand it was built this way for the global view, but it's strange since the dropdown belongs to core actions, and has nothing to do with global. Is it not that the tab looks "active" like the WP logo? I'd be happy to follow up on this on a different PR (cc. @matt-west) 

![screen 2025-03-04 at 15 42 47](https://github.com/user-attachments/assets/c81ddb0d-0843-4fe5-99ad-aa31d27399d1)

Otherwise, we should to tweak the badges to adapt on the grey background.

![screen 2025-03-04 at 15 39 07](https://github.com/user-attachments/assets/551beb7f-057f-4caa-bb0f-f370e1466b7d)

nuriapenya commented:
@youknowriad fantastic, let's do this for now. As a sum up of the changes:

* No dot with colors 
* No 'Updates' badge
* Style variation for global level 

![screen 2025-03-05 at 08 44 06](https://github.com/user-attachments/assets/07df6443-d438-44e9-85bb-99a7b472afb7)
![image](https://github.com/user-attachments/assets/3ec934f9-4991-4d70-82c4-a3176dd8b803)


FYI I shared an alternative with the Site Icon + showing the URL here in slack: p1741162725855429/1741089331.046709-slack-C9EJ7KSGH, since Matias asked for a version with that info. I think this could work very well to match the info from the site card, but it deviates from the current core's topbar. WDYT?

![image](https://github.com/user-attachments/assets/87144ab6-04ca-408b-b8a0-9521c3c272c9)

tyxla commented on client/layout/masterbar/logged-in.jsx:
```diff
@@ -400,6 +407,139 @@ class MasterbarLoggedIn extends Component {
+		if ( isP2Hub ) {
+			badges.push(
+				<span key="p2-workspace" className="masterbar__site-badge is-p2-workspace">
+					P2 Workspace
```
"p2" and "p2 workspace" seem odd as statuses. Will they ever change for a site? What value do they provide?

youknowriad replied:
These are copied from what we currently have. I don't really understand "p2 workspace"

tyxla replied:
I see. p2 workspace is like a network - it allows you to have multiple p2s sites, almost emulating the multisite - single site model.

tyxla commented on client/layout/masterbar/logged-in.jsx:
```diff
@@ -427,13 +568,41 @@ class MasterbarLoggedIn extends Component {
+									<div className="masterbar__info-badges">{ siteBadges }</div>
+								</div>
+							) }
+							<div className="masterbar__site-info">
+								<span className="masterbar__site-info-label">{ translate( 'Plan' ) }</span>
+								<div className="masterbar__info-badges">
+									<span className="masterbar__site-badge is-plan">{ sitePlanName }</span>
+								</div>
+							</div>
```
My perspective as a user: I've never understood why we need to surface the current plan so visibly. My assumption is that users will rarely change their plan, and they have dedicated places to see this information. With that in mind, I don't understand what value displaying the Plan here (or in the sidebar) brings.

richtabor replied:
I think it's good to surface, strategically—especially if it's the free plan—as It's a hint to what you can get. 

We should explore a more consistent upsell strategy, perhaps like Linear or Slack does: where every time you interact with something indicative of your current plan, you see the same upsell.

youknowriad committed:
Tweak design

youknowriad committed:
Use CoreBadge component

youknowriad commented:
@nuriapenya I updated the design.

1- I've used The Core [Badge](https://wordpress.github.io/gutenberg/?path=/docs/components-badge--docs) component. But to match the mockups, I had to override the colors of the component, which is not the ideal. Is there any reason to not use the default core badge colors?
2- The gray colors that I've used don't match perfectly with the mockup, the reason is that we do have gray color variables and they were being used in several places, so I thought I'd reuse these. I didn't find variables that match the mockups directly. WDYT? (cc @ciampo in case I'm missing something about the gray variables in Calypso)
3- I didn't touch the design of the links (View Site, My Home) because these menu items's design match all the menu items design in the admin bar, I only focused on designing the badges area.
4- On the alternative design, it should be doable here in Calypso, but harder when it comes to applying in WP-Admin (which we will need to do for people that choose the "classic view"). I'm happy to give it a try though, if we're convinced this is the way to go.

ciampo commented on client/layout/masterbar/item.tsx:
```diff
@@ -137,6 +137,7 @@ class MasterbarItem extends Component< MasterbarItemProps > {
 								{ item.label }
 							</a>
 						) }
+						{ ! item.onClick && ! item.url && item.label }
```
It may be better to wrap them in a `<span />` for better "consistency" and potentially less layout differences with the other item types (`<button />` and `<a />`) ?

ciampo commented:
> 2- The gray colors that I've used don't match perfectly with the mockup, the reason is that we do have gray color variables and they were being used in several places, so I thought I'd reuse these. I didn't find variables that match the mockups directly. WDYT? (cc @ciampo in case I'm missing something about the gray variables in Calypso)

From what I understand, the "global" color variables should come from `@automattic/color-studio` — I wonder if we should use those variables, even if they differ slightly form the mockups, instead of adding more `--color-global-masterbar-site-*` variables? And in general, I wonder if we should aim are removing also existing masterbar-specific variables in favour of color-studio variables? @jasmussen and @matt-west may know best.

youknowriad commented:
> --color-global-masterbar-site-* variables? 

I'm already using the studio colors, the only reason I'm using the master-* colors is to override the colors in the "global view" which was already done this way, I'm just following the way it was done for other parts of the master bar. This can be "removed" once either the master bar is removed from the global view or made consistent with the rest.

jasmussen commented:
I'm increasingly cold on color-studio. I do agree we should ideally use variables, and perhaps it's going to be messy if we start mixing what exists with the sass variables you can get from importing the `base-styles` package from core. But at the same time, that package has 8 shades of gray, and they're sass variables which means they're compiled. Color Studio loads 700 CSS variables, twice, on every page where it's included. I realize, that's already every page on wp.com, but as far as reducing entropy, the path forward that I see is to stop using color studio. Related: p1741173086517169-slack-C9EJ7KSGH

ciampo commented:
Thank you @jasmussen for sharing the context. Nothing blocking for this PR specifically, but once we've done with the [color studio update](https://github.com/Automattic/wp-calypso/pull/99580) and any follow-ups, we could take a moment to assess the best strategy there.

youknowriad commented:
@nuriapenya Did you check the latest state of this PR, would love your input.

nuriapenya commented:
@youknowriad sorry for the delay 🙏🏼 it looks fantastic. Couple of things I found:

* I'm still not sure why we show the "Updates" badge. Is it needed? Usually, when there are updates to be done we already have an item in the navigation to say so.
<img width="486" alt="image" src="https://github.com/user-attachments/assets/80a073d5-7ee9-4bf0-bd21-8e7ec6ad3454" />

* Stating sites do not have a plan per se. Perhaps we should remove this info? Might be misleading:
<img width="952" alt="scren2" src="https://github.com/user-attachments/assets/5e31cb59-1224-4a52-8242-ee92f1f58eb4" />

* We also have the case when a plan has expired. Perhaps we need to find a better way to communicate that. Not to block this PR, this can be done in a follow up.
<img width="942" alt="screenshot" src="https://github.com/user-attachments/assets/6a29917f-0560-4852-b6f7-1e57dad2e65a" />

Regarding the UI points:

> 1- I've used The Core [Badge](https://wordpress.github.io/gutenberg/?path=/docs/components-badge--docs) component. But to match the mockups, I had to override the colors of the component, which is not the ideal. Is there any reason to not use the default core badge colors?
2- The gray colors that I've used don't match perfectly with the mockup, the reason is that we do have gray color variables and they were being used in several places, so I thought I'd reuse these. I didn't find variables that match the mockups directly. WDYT?

I'm not sure I follow, but I used exactly the same badge with Core's colors. Perhaps I never shared the Figma file? lPSxx2wELvTBINL9NNGIW4-fi-148_8045 The only thing I did, since there is no "dark mode" was using a different set of darker grays for the site-level version. But those grays are part of Core's gray color variables. (Sorry, I'm a bit newbie on this area, perhaps I did not use the correct things).

<img width="1391" alt="mocks" src="https://github.com/user-attachments/assets/9af0c3a9-6e9d-4a5a-9b8f-f600e9cf1979" />


> 3- I didn't touch the design of the links (View Site, My Home) because these menu items's design match all the menu items design in the admin bar, I only focused on designing the badges area.
Perfect 👌🏼 

> 4- On the alternative design, it should be doable here in Calypso, but harder when it comes to applying in WP-Admin (which we will need to do for people that choose the "classic view"). I'm happy to give it a try though, if we're convinced this is the way to go.
No need for now. We can try for another iteration. There are a lot of things we need to figure out for Hosting vs. Self-Serve customers first.

youknowriad committed:
Tweaks

youknowriad commented:
In the last commit, I removed the "updates" and removed the plan from staging sites.

@nuriapenya on the colors, it was more of a technical question, I'm not sure how to use the "core gray colors" on Calypso unless I hard code the values. Do we have Sass variables or CSS variables for these? I have no idea. I've used "studio colors" which don't match 100% with the mockup.

jasmussen commented:
You should be able to import the `base-styles` package, which comes with the core colors. @matt-west in case you have time to respond, you've also some color insight, just in case you have clarity.

youknowriad committed:
Update the colors to use core colors

youknowriad committed:
Small tweak

youknowriad commented:
How should the background of the badge area adapt to the user color scheme? Ideally I would say we should illuminate the background and just have the same background as the "view site", "my home" links. Would that be an option design wise?

Right now the user color schemes don't provide a "dark admin bar color".

youknowriad commented:
<img width="276" alt="Screenshot 2025-03-17 at 11 49 39 AM" src="https://github.com/user-attachments/assets/0b089c6b-41d3-4da7-be97-4b9b9bb48247" />

jasmussen commented:
> How should the background of the badge area adapt to the user color scheme? Ideally I would say we should illuminate the background and just have the same background as the "view site", "my home" links. Would that be an option design wise?

Can you do this?

```
[badge] {
	color: inherit;
	background-color: color-mix( in srgb, currentColor 10%, transparent );
}
```

This should make the background the same color as the text, but 10% opaque.

youknowriad commented:
@jasmussen that seems to create an inconsistent result between Calypso and wp-admin for some reason. 

<img width="268" alt="Screenshot 2025-03-17 at 12 02 48 PM" src="https://github.com/user-attachments/assets/a61d7366-f444-4b5a-8d82-eddd22dd6d06" />
<img width="270" alt="Screenshot 2025-03-17 at 12 02 57 PM" src="https://github.com/user-attachments/assets/597ac632-3795-47fb-9c7e-5b9ee7754ce7" />

youknowriad commented:
I may be able to make it work with hacky CSS, let's see.

jasmussen commented:
Took a quick look. The main problem is that you are running into the Calypso color schemes, which are not in a good place. Rather than spend a lot of time getting those to work (trust me, it's a super deep rabbit hole), I would recommend you look at color-mix yet again. For instance, here's Sunrise in your PR:

![Screenshot 2025-03-17 at 12 57 48](https://github.com/user-attachments/assets/06537ffd-baa6-4417-af28-d5b7b04eb07c)

It looks poor because the CSS variables for that subsection are not correct. Instead of getting those to work, why don't you colorize the subsection like this?

```
color-mix(in srgb, var(--color-masterbar-item-hover-background), black 10% )
```

That takes the flyout menu bg color, makes it 10% darker, and you get this:

![Screenshot 2025-03-17 at 13 00 13](https://github.com/user-attachments/assets/d91ea6cd-7458-46cc-8e06-2260e1dab274)

You'd need to re-do the chip again. 

But overall, I'd recommend: don't use studio variables, if you do you have to update 18 color schemes, most of them I'd like to retire anyway.

youknowriad commented:
@jasmussen With what value you would replace `--color-masterbar-item-hover-background` in wp-admin?

youknowriad commented:
I'm working on a companion PR to this one, to bring these updates to wp-admin's admin bar for both atomic and simple sites here. https://github.com/Automattic/jetpack/pull/42499

jasmussen commented:
> With what value you would replace --color-masterbar-item-hover-background in wp-admin?

That's the thing, this variable is the background color of the hover menu. So you keep that one as is, and just make it darker using the color-mix trick. WDYT?

youknowriad commented:
@jasmussen That variable is not usable in wp-admin, it doesn't exist there at all, it only exists in Calypso so we can't use it there.

jasmussen commented:
Oh, my mistake, yes, this is a mess, I hope we can retire some of those non-wp-admin pages soon so we don't have to juggle those values.

I'm curious if @Automattic/dotcom-design has any thoughts here. I'm tempted to override the FULL color of all those dropdowns, so the dropdowns themselves are _never_ styled. That would let us reach Núrias "dark" design for every scolor scheme. This would be its own type of mess, though, so not sure.

jameskoster commented:
> With what value you would replace --color-masterbar-item-hover-background in wp-admin?

I don't know if it helps, but the inspiration for the colorisation was what we find in the 'About' menu in wp-admin, this one:

<img width="171" alt="Screenshot 2025-03-18 at 14 12 20" src="https://github.com/user-attachments/assets/6655f05b-5a40-4441-9c32-df3d2a58c248" />

Can we not re-use those classes in wp-admin?

youknowriad commented:
<img width="269" alt="Screenshot 2025-03-19 at 9 36 46 AM" src="https://github.com/user-attachments/assets/92160454-b43c-413a-af14-ef80d9d8c987" />

<img width="266" alt="Screenshot 2025-03-19 at 9 36 57 AM" src="https://github.com/user-attachments/assets/48863e9a-5d30-42d2-b324-0e9830ca6aac" />

@jameskoster Thanks, I've been able to use that in wp-admin, it gives the following result (screenshots above), I had to update the badge color slightly otherwise it doesn't show up in the default scheme. If you think that's good, I can update the current PR as well (calypso)

ciampo commented:
> Connecting some dots, although there's talk of removing the widget in question, I wonder if the badge could use the same format as this one: [#101285 (comment)](https://github.com/Automattic/wp-calypso/issues/101285#issuecomment-2728213030)
> 
> That is, outline style badge, rather than solid.

It depends on what underlying "badge" implementation we want to use. If we want to align to Gutenberg more, then we should use the [`Badge` component](https://wordpress.github.io/gutenberg/?path=/docs/components-badge--docs) from `@wordpress/components`. For example, @mirka has recently [refactored some A4A UI to use the Core badge component](https://github.com/Automattic/wp-calypso/pull/101376).

See also [this discussion](https://github.com/Automattic/wp-calypso/pull/100764#discussion_r1979758290) for an overview of all badge components that are around Calypso)

youknowriad committed:
Match wp-admin style

youknowriad commented:
I updated the colors in this PR to match approximatively what we have in WP-Admin, that classname doesn't exist in Calypso so I used the color-mix technique to achieve a close result. 

For the badge, I'm happy to update but yeah, there's no reusable badge for that outlined one. We'd be creating an exception.

jameskoster commented:
I'd vote for the Coming soon badge aligning with the Badge component in wp-components rather than the other way around. But since the site card is being removed I don't think it matters too much.

That said, we'll likely need to create a style exception anyway because Badge was not designed to sit on a dark background 😅 Could we set the text color to inherit from the menu, then use a semi-transparent white background? Not ideal, but not the worst overrides in the world.

Quick Inspector hack:

<img width="253" alt="Screenshot 2025-03-19 at 15 30 29" src="https://github.com/user-attachments/assets/88a3abfd-c332-4420-afaf-9a8133d2487f" />
<img width="241" alt="Screenshot 2025-03-19 at 15 31 41" src="https://github.com/user-attachments/assets/837cb7c9-efd5-475e-8324-86898592c302" />
<img width="252" alt="Screenshot 2025-03-19 at 15 32 02" src="https://github.com/user-attachments/assets/80cdcb1c-fc40-41f1-9d1f-d7e2a3a04ab7" />

youknowriad commented:
@jameskoster I'm happy to make the change assuming you all agree. I don't want to do a lot of back and forth though as it involves multiple PRs. Can we just agree on something and move forward.

crisbusquets commented:
> Could we set the text color to inherit from the menu, then use a semi-transparent white background? Not ideal, but not the worst overrides in the world.

I totally missed the second badge ("Badge") the three times I read this thread 😅 It could work if we adjust the semi-transparent background a little bit. But then we might have a11y issues.

jameskoster commented:
<!-- linear:replyTo:2657217124 -->

Iirc that was 10% opacity. We can bump it up to whatever so long as we avoid contrast issues like you said. I really don't have a strong opinion about that.

youknowriad commented:
Hey folks, this is blocked for too long. What's the next steps here:

 - Should I use opacity for badges like @jameskoster suggested
 - Should I revert to the old badge design like @jasmussen suggested 
 - Also, are we ok with this PR or should we explore the one showing the site icon instead of the "home" icon separately before landing this. cc @nuriapenya

jasmussen commented:
I'm personally happy with either, honestly. Because it's in the adminbar here, I'm tempted to just pick what is simplest to work with, and move forward.

nuriapenya commented:
> Also, are we ok with this PR or should we explore the one showing the site icon instead of the "home" icon separately before landing this. 

I don't think it's necessary for this PR, since there's a parallel work happening around hosting dashboard which might affect this area.

@youknowriad  I just re-tested the current PR and to me, what you have now looks and works totally fine. I'd say to go for it, and if we need to revise the badge style, we could do it on a separate issue (and more holistically).

youknowriad committed:
Update badge colors

youknowriad commented:
I updated the badge colors to use @jameskoster's approach.

I would love an approval here.

youknowriad commented:
Thanks @jasmussen I'm going to wait for Monday to merge here, because this basically removes the "view" pages from Calypso which could potentially be impactful (although I don't really expect it to be the case, but better be careful)
